### DISCUSSION TITLE
why nextjs14 cant render charts?

### DISCUSSION THREAD

**Discussion by flog89** [community/community#134924] [Repositories]
### Select Topic Area

Question

### Body

why nextjs14 cant render charts in shadcn

**Comment by Abdulrehmanatif** [comment]
Next.js 14 itself shouldn’t prevent charts from rendering, so if you're having trouble, it might be due to several reasons. First, make sure that the charting library you’re using is compatible with Next.js 14. Sometimes updates to either the framework or the library can lead to compatibility issues.

It’s also worth checking your Next.js configuration to ensure it’s set up correctly for client-side rendering, which is often necessary for charting libraries to work properly. Make sure that all dependencies for the charting library are correctly installed and up-to-date.

Additionally, review your code implementation to ensure you’re using the charting library’s API correctly and providing all the necessary props or configurations. If there are any error messages in your console or server logs, they can provide valuable clues about what might be going wrong.

Lastly, if Shadcn has specific requirements or known issues, it could be helpful to consult its documentation or community forums for any related problems or solutions. If you have more details about the specific charting library or the issues you’re facing, I can offer more targeted advice.

**Comment by mu6m** [comment]
upgrade to nextjs15
update your package.json to have 
```
    "next": "^15.0.0-rc.0", 
```
dont use the 14.2.5

**Comment by ebndev** [comment]
Please see [this comment](https://github.com/orgs/community/discussions/135204#discussioncomment-10288226) for reference to why this discussion was closed and locked. Thanks!
